crates/edr_provider/tests/eip7702.rs

```diff
@@ -0,0 +1,361 @@
+#![cfg(feature = "test-utils")]
```

Comment:
It's difficult for me to see if all the test cases from the design doc are covered in this module. Can you please clarify if they are covered?